avm/res/network/azure-firewall/tests/e2e/tunneling/main.test.bicep

```diff
@@ -0,0 +1,78 @@
+targetScope = 'subscription'
+
+metadata name = 'Forced tunneling'
+metadata description = 'This instance deploys the module and sets up forced tunneling.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.azurefirewalls-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'naftunn'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    location: resourceLocation
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    publicIPName: 'dep-${namePrefix}-pip-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      location: resourceLocation
+      name: '${namePrefix}${serviceShort}001'
+      virtualNetworkResourceId: nestedDependencies.outputs.virtualNetworkResourceId
+      additionalPublicIpConfigurations: [
+        {
+          name: 'ipConfig01'
+          publicIPAddressResourceId: nestedDependencies.outputs.publicIPResourceId
+        }
+      ]
+      azureSkuTier: 'Standard'
+      enableForcedTunneling: true
+      managementIPAddressObject: {
+        publicIPAllocationMethod: 'Static'
+        roleAssignments: [
+          {
+            roleDefinitionIdOrName: 'Reader'
+            principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+            principalType: 'ServicePrincipal'
+          }
+        ]
```

Comment:
Why the role assignment? I'd suggest to keep the test focused on the features you want to test to not accidently imply that e.g. a role assignment would need to be done in order to deploy forced tunneling

Reply:
Updated.